addons/account/models/company.py

```diff
@@ -60,10 +60,14 @@ class ResCompany(models.Model):
         ('round_globally', 'Round Globally'),
         ], default='round_per_line', string='Tax Calculation Rounding Method')
     currency_exchange_journal_id = fields.Many2one('account.journal', string="Exchange Gain or Loss Journal", domain=[('type', '=', 'general')])
-    income_currency_exchange_account_id = fields.Many2one('account.account', related='currency_exchange_journal_id.default_credit_account_id', readonly=False,
-        string="Gain Exchange Rate Account", domain="[('internal_type', '=', 'other'), ('deprecated', '=', False), ('company_id', '=', id)]")
-    expense_currency_exchange_account_id = fields.Many2one('account.account', related='currency_exchange_journal_id.default_debit_account_id', readonly=False,
-        string="Loss Exchange Rate Account", domain="[('internal_type', '=', 'other'), ('deprecated', '=', False), ('company_id', '=', id)]")
+    income_currency_exchange_account_id = fields.Many2one(
+        'account.account',
+        string="Gain Exchange Rate Account",
+        domain="[('internal_type', '=', 'other'), ('deprecated', '=', False), ('company_id', '=', id)]")
+    expense_currency_exchange_account_id = fields.Many2one(
+        'account.account',
+        string="Loss Exchange Rate Account",
+        domain="[('internal_type', '=', 'other'), ('deprecated', '=', False), ('company_id', '=', id)]")
```

Comment:
thks for the indent! :)
To be perfect, you could add `comodel_name=` juste before `'account.account'` :p